price/fixed_method_pricing.go

```diff
@@ -0,0 +1,42 @@
+package price
+
+import (
+	"fmt"
+	"github.com/singnet/snet-daemon/blockchain"
+	"github.com/singnet/snet-daemon/handler"
+	"math/big"
+)
+
+type FixedMethodPrice struct {
+	//Service/Method is the key and value is he price
+	methodToPriceMap map[string]*big.Int
+}
+
+func (priceType FixedMethodPrice) GetPrice(GrpcContext *handler.GrpcStreamContext) (price *big.Int, err error) {
```

Comment:
Should it be:
```suggestion
func (priceType *FixedMethodPrice) GetPrice(GrpcContext *handler.GrpcStreamContext) (price *big.Int, err error) {
```
?

Reply:
This one was intentionally left without a * ( we would pass a star ideally when a value is to be changed and reflected back at the calling location.
We may need to check the uses of * across daemon for consistency probably start fixing as we touch those classes again.

Reply:
Ok, I see your point, I am not sure if `Go` compiler will wisely optimize it and make a copy of the  object only when object is changed. I mean in `Go` it is easy to make error like one you have fixed in `cmd.Serve.go`: one developer created method without `*` because object is not modified, and another developer added modifying operation and at the end we have copy of the object and change is not applied to main instance. Unit testing solves it but is our unit test coverage ideal?

Probably we should find some `Golang` best practices on this. Do you have a link?

Reply:
Well, I have found couple of links:
- https://flaviocopes.com/golang-methods-receivers/
- https://github.com/golang/go/wiki/CodeReviewComments#receiver-type
- https://golang.org/doc/effective_go.html#pointers_vs_values

Using logic of this links it is preferable to use value receiver here as it is a small object and method doesn't modifies it. I agree it is safer from threading perspective. But I still think that `Golang` ability to silently compile the call of the method using pointer or value can lead to errors like we had in `snet.cmd.serve.go`.

Summary: lets' keep it as is.